tests/unit/test_shadow_account_policy.py

```diff
@@ -0,0 +1,486 @@
+    }, 'Minimum password age must be lower than the maximum password age'),
+))
+def test__account_policy_validation(sec, error, set_fips_available):
+    with Client() as c:
+        with pytest.raises(Exception, match=error):
+            c.call('system.security.update', sec, job=True)
+
+
+def test__password_history(password_history):
+    """ Verify that password history is properly applied """
+    with create_user('history_user', random_password=False, password='Canary') as u:
+        with Client() as c:
+            with pytest.raises(Exception, match='requires a password that does not match'):
+                c.call('user.update', u['id'], {'password': 'Canary'})
+
+            for i in range(0, TEST_HISTORY):
+                c.call('user.update', u['id'], {'password': f'Canary{i}'})
+
+            # We've cycled through history and should be OK again
+            c.call('user.update', u['id'], {'password': 'Canary'})
+
+
+def test__password_expired(old_admin):
+    """ Verify that expired password gives EXPIRED response type """
+    with Client('ws://127.0.0.1/api/current') as c:
+        resp = c.call('auth.login_ex', {
+            'mechanism': 'PASSWORD_PLAIN',
+            'username': old_admin['username'],
+            'password': old_admin['password']
+        })
+        assert resp['response_type'] == 'SUCCESS'
+
+    with security_config(max_password_age=60):
+        with Client() as c:
+            user = c.call('user.get_instance', old_admin['id'])
+            assert user['password_change_required']
+
+            # expired accounts should generate an alert
+            alerts = c.call('alert.run_source', 'SecurityLocalUserAccountExpiration')
+            assert any([alert['klass'] == 'LocalAccountExpired' for alert in alerts]), str(alerts)
+
+        with Client('ws://127.0.0.1/api/current') as c:
+            resp = c.call('auth.login_ex', {
+                'mechanism': 'PASSWORD_PLAIN',
+                'username': old_admin['username'],
+                'password': old_admin['password']
+            })
+            assert resp['response_type'] == 'EXPIRED'
+
+
+def test__password_expiry_warning(old_admin):
+    """ Verify that account attributes show PASSWORD_CHANGE_REQUIRED """
+    with Client('ws://127.0.0.1/api/current') as c:
+        resp = c.call('auth.login_ex', {
+            'mechanism': 'PASSWORD_PLAIN',
+            'username': old_admin['username'],
+            'password': old_admin['password']
+        })
+        assert resp['response_type'] == 'SUCCESS'
+
+    with security_config(max_password_age=10) as sec:
+        with Client() as c:
+            c.call('datastore.update', 'account.bsdusers', old_admin['id'], {
+                'bsdusr_last_password_change': SEVEN_DAYS_AGO
+            })
+            c.call('etc.generate', 'shadow')
+            user = c.call('user.get_instance', old_admin['id'])
+            assert user['password_change_required'], str({"user": user, "sec": sec})
```

Comment:
Similar to previous comment,  Before we set the `max_password_age` above, it might be an idea to ensure `password_change_required` is **not** set.  (Admittedly, somewhat mitigated by what follows.)

(Equally, ensure could ensure no LocalAccountExpiring to start.)